events/source/EventQueue.cpp

```diff
@@ -23,10 +23,16 @@ namespace events {
 
 EventQueue::EventQueue(unsigned event_size, unsigned char *event_pointer)
 {
-    if (!event_pointer) {
-        equeue_create(&_equeue, event_size);
+    if (event_size == 0) {
+        // 1B dummy buffer for static queue, when queue initialized as EventQueue(0)
+        static uint8_t _dummy_buffer;
+        equeue_create_inplace(&_equeue, 1, &_dummy_buffer);
```

Comment:
Still nervous about what it is that you end up accessing - why was the alignment a problem? Does it do a memset or something?

Reply:
It must be clarified, working on it...

Reply:
This is the error I was talking about when do the align on 1B  buffor and run queue test
`mbed test -t GCC_ARM -m k64f -n tests-events-queue -v`
```
[1567090373.94][CONN][RXD] >>> Running case #16: 'Testing static events queue'...
[1567090374.00][CONN][RXD]
[1567090374.00][CONN][RXD]
[1567090374.00][CONN][INF] found KV pair in stream: {{__testcase_start;Testing static events queue}}, queued...
[1567090374.02][CONN][RXD] ++ MbedOS Error Info ++
[1567090374.06][CONN][RXD] Error Status: 0x80FF0144 Code: 324 Module: 255
[1567090374.14][CONN][RXD] Error Message: Assertion failed: p == reinterpret_cast<T *>(&_data)
[1567090374.16][CONN][RXD] Location: 0x8109
[1567090374.20][CONN][RXD] File: ./platform/SingletonPtr.h+139
[1567090374.22][CONN][RXD] Error Value: 0x0
[1567090374.32][CONN][RXD] Current Thread: main  Id: 0x20001230 Entry: 0x8D93 StackSize: 0x1000 StackMem: 0x20002300 SP: 0x20002FE0
[1567090374.40][CONN][RXD] For more info, visit: https://mbed.com/s/error?error=0x80FF0144&tgt=K64F
[1567090374.42][CONN][RXD] -- MbedOS Error Info --
[1567090374.42][CONN][RXD]
[1567090374.48][CONN][RXD] = System will be rebooted due to a fatal error =
[1567090374.54][CONN][WRN] UnicodeDecodeError encountered!
[1567090385.67][HTST][INF] test suite run finished after 20.14 sec...
```

Reply:
Yes, but probably all that signifies it that a SingletonPtr was placed near your _dummy in the ROM, and you accidentally overwrote the SingletonPtr. Which means it must be doing something with that buffer. A watchpoint would indicate where that's happening

Reply:
I traced the problem
The size alignment was doing `1 -> 4294967292`
https://github.com/ARMmbed/mbed-os/blob/ea582f3e11023482f17736d32a40ebd2eb899a96/events/source/equeue.c#L68-L69

Then to test that dynamic event allocation will fail my test is doing this:
```c++
static EventQueue g_queue(0);
Event<void()> e0 = g_queue.event(func0);
TEST_ASSERT_EQUAL(0, e0.post());
```
Actually it should do nothing but as the queue size was overwritten by alignment (`1 -> big number`) it returns some address around our `dummy_buffer`

So actually the fix is out there
```c++
int equeue_create_inplace(equeue_t *q, size_t size, void *buffer)
{
    // setup queue around provided buffer
    // ensure buffer and size are aligned
    if (size >= sizeof(void *)) {
        q->buffer = (void *)(((uintptr_t) buffer + sizeof(void *) -1) & ~(sizeof(void *) -1));
        size -= (char *) q->buffer - (char *) buffer;
        size &= ~(sizeof(void *) -1);
    } else {
        // don't align when size less then pointer size
        // e.g. static queue (size == 1)
        q->buffer = buffer;
    }
```

Reply:
@kjbracey-arm 
Also changed `dummy -> this`
```c++
EventQueue::EventQueue(unsigned event_size, unsigned char *event_pointer)
{
    if (event_size == 0) {
        // As static queue (EventQueue(0)) won't perform any access to its dummy buffer
        // set 1B dummy buffer as pointer to itself
        equeue_create_inplace(&_equeue, 1, this);
```

Reply:
If we're confident it's not accessing the buffer, why not `0, nullptr`? If it needs 1 byte, you'd need to do something, but 0, nullptr is fine if it isn't accessing it, and clearer.